src/cursor/common.rs

```diff
@@ -37,17 +37,15 @@ where
     provider: P,
     client: Client,
     info: CursorInformation,
-    buffer: VecDeque<RawDocumentBuf>,
```

Comment:
I moved the stuff that needed to be preserved between `GenericCursor` constructions in a `CursorState` struct to ensure nothing gets forgotten.